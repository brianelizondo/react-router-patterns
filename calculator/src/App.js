import {React, useState } from 'react';
import { Routes, Route } from "react-router-dom";
import './App.css';

import Add from "./Add";
import Subtract from "./Subtract";
import Multiply from "./Multiply";
import Divide from "./Divide";

import Calculator from './Calculator';

function App() {
    return (
        // REACT ROUTER CALCULATOR WITH DIFFERENTS COMPONENTS
        // <div className="App">
        //     <Routes>
        //         <Route path="/add/:a/:b" element={ <Add /> } />
        //         <Route path="/subtract/:a/:b" element={ <Subtract /> } />
        //         <Route path="/multiply/:a/:b" element={ <Multiply /> } />
        //         <Route path="/divide/:a/:b" element={ <Divide /> } />
        //     </Routes>
        // </div>

        <div className="App">
            <Routes>
                <Route path="/:operation/:a/:b" element={ <Calculator /> } />
            </Routes>
        </div>
    );
}

export default App;
