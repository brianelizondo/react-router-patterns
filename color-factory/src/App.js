import {React, useState } from 'react';
import { Routes, Route, Navigate } from "react-router-dom";
import './App.css';

import ColorList from "./ColorList";
import ColorDetails from "./ColorDetails";
import ColorNew from "./ColorNew";

function App() {
    const INITIAL_STATE = [];
    const [colors, setColors] = useState(INITIAL_STATE);

    const addColor = newColor => {
        setColors(colors => [newColor, ...colors]);
    };

    return (
        <div className="App">
            <Routes>
                <Route path="/colors/new" element={ <ColorNew addColor={addColor} /> } />
                <Route path="/colors/:color" element={ <ColorDetails colors={colors} /> } />
                <Route path="/colors" element={ <ColorList colors={colors} /> } />
            </Routes>
        </div>
    );
}

export default App;
