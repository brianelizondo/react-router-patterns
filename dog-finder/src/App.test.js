import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import App from './App';
import whiskey from "./whiskey.jpg";
const dogTest = [{
    name: "Whiskey",
    age: 5,
    src: whiskey,
    facts: [
        "Whiskey loves eating popcorn.",
        "Whiskey is a terrible guard dog.",
        "Whiskey wants to cuddle with you!"
    ]
}];

// Smoke Tests
it("'App' component renders without crashing", () => {
    render(<MemoryRouter><App /></MemoryRouter>);
});

// Snapshot Tests
it("'App' matches snapshot", () => {
    const {asFragment} = render(<MemoryRouter><App /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});

// Others test
test("/dogs route", () => {
    const { getByText } = render((
        <MemoryRouter initialEntries={["/dogs"]}>
            <App />
        </MemoryRouter>
    ));
    
    expect(getByText("PET's LIST:")).toBeInTheDocument();
});

test("/dogs/:name route", () => {
    const { getByAltText } = render((
        <MemoryRouter initialEntries={["/dogs/whiskey"]}>
            <App dogs={dogTest} />
        </MemoryRouter>
    ));
    
    expect(getByAltText("Whiskey")).toBeInTheDocument();
});

test("NavBar links", () => {
    const { getByText, getByAltText } = render((
        <MemoryRouter initialEntries={["/dogs"]}>
            <App dogs={dogTest} />
        </MemoryRouter>
    ));
    expect(getByText("PET's LIST:")).toBeInTheDocument();

    const link = getByText("Home");
    fireEvent.click(link);
    expect(getByText("PET's LIST:")).toBeInTheDocument();
});