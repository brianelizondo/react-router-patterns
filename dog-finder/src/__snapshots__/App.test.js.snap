// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'App' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <nav
      class="NavBar"
    >
      <a
        aria-current="page"
        class="NavBar-active"
        href="/dogs"
      >
        Home
      </a>
      <a
        href="/dogs/whiskey"
      >
        Whiskey
      </a>
      <a
        href="/dogs/duke"
      >
        Duke
      </a>
      <a
        href="/dogs/perry"
      >
        Perry
      </a>
      <a
        href="/dogs/tubby"
      >
        Tubby
      </a>
    </nav>
    <div
      class="DogList"
    >
      <div
        class="DogList-logo"
      >
        <img
          alt=""
          src="dog_finder_logo.jpg"
        />
      </div>
      <div
        class="DogList-list"
      >
        <h2>
          PET's LIST:
        </h2>
        <div>
          <a
            href="/dogs/whiskey"
          >
            Whiskey
          </a>
        </div>
        <div>
          <a
            href="/dogs/duke"
          >
            Duke
          </a>
        </div>
        <div>
          <a
            href="/dogs/perry"
          >
            Perry
          </a>
        </div>
        <div>
          <a
            href="/dogs/tubby"
          >
            Tubby
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
